project_electron/orgs/models.py

```diff
@@ -315,3 +339,66 @@ def log_it(cls, code, archive=None):
 
     class Meta:
         ordering = ['-created_time']
+
+class BagInfoMetadata(models.Model):
+    archive =                       models.ForeignKey(Archives, blank=True,null=True)
+    source_organization =           models.ForeignKey(Organization, blank=True,null=True)
+    external_identifier =           models.CharField(max_length=256)
+    internal_sender_description =   models.TextField()
+    title =                         models.CharField(max_length=256)
+    date_start =                    models.DateTimeField()
+    date_end =                      models.DateTimeField()
+    record_creators =               models.ManyToManyField(RecordCreators, blank=True)
+    record_type =                   models.CharField(max_length=30)
+    language =                      models.ManyToManyField(LanguageCode, blank=True)
+    bagging_date =                  models.DateTimeField()
+    bag_count =                     models.CharField(max_length=10)
+    bag_group_identifier =          models.CharField(max_length=256)
+    payload_oxum =                  models.CharField(max_length=20)
+    bagit_profile_identifier =      models.URLField()
+
+    @classmethod
+    def save_metadata(cls, metadata, archive=None):
```

Comment:
- This would be more useful as a method of Archives just like get_bag_data should be a save_bag_data, that way you don't need to expose another object, and prevents class method.
- also, foreign key shouldn't be blank or null

Reply:
-And if logic needs to be in try/catch then return bool so code can react to it.